test/bin/pyutils/common.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import os
+import pathlib
+import sys
+import subprocess
+from typing import List
+
+
+def should_skip(file):
+    # Implement your skipping logic here
+    return False
+
+
+def record_junit(groupdir, containerfile, filetype, status):
+    # Implement your recording logic here
+    pass
+
+
+def print_msg(msg: str, file=sys.stderr):
+    print(msg, file=file)
+
+
+def get_env_var(var_name: str):
+    """Get an environment variable or exit if not set."""
+    value = os.environ.get(var_name)
+    if value is None:
+        print_msg(f"Error: {var_name} environment variable not set.")
+        sys.exit(1)
+    return value
+
+
+def run_command(command: List[str], dry_run: bool):
+    """Run the command or print the command line depending on the dry run argument"""
+    if dry_run:
+        print_msg(f"[DRY RUN] {' '.join(command)}")
+        return None
+
+    print_msg(f"[RUN] {' '.join(command)}")
+    return subprocess.run(command, check=True)
+
+
+def run_command_in_shell(command: str):
+    """Run the command through shell and return its output"""
+    print_msg(f"[SHELL] {command}")
+    result = subprocess.run(
+        command,
+        check=True, shell=True, text=True,
+        env=os.environ.copy(), stdout=subprocess.PIPE)
+    return result.stdout.strip()
+
+
+def create_dir(dir: str, chdir: bool):
+    """Attempt recursive directory creation and optionally change current directory"""
+    path = pathlib.Path(dir)
+    path.mkdir(parents=True, exist_ok=True)
+    if chdir:
+        os.chdir(path)
```

Comment:
I'm not fan of this especially without accompanying "popd`. What do you think about using full absolute paths whenever possible - it's verbose, but there's no "context" around the command with relative path.